### PR TITLE
DBZ-3611 Avoid NPE with database include/exclude lists

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3611

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Hey, I don't quite understand that fix yet. What is triggering the NPE mentioned in the issue comments? In particular, I'd have expected that these unsupported props won't be touched whatsoever by this connector (which doesn't support them atm.), but you seem to indicate that their sole presence in the actual config will trigger some NPE?

**Comment by Naros** [comment]
So the default behavior in `JdbcConnection#readSchema` is the problem.

This method relies solely on the database driver to correctly return the catalog, schema, and table names but unfortunately the Oracle driver defaults to always turning the catalog name as `null`.  This means when we construct the `TableId` from this tuple and then verify if the tuple passes the `TableFilter` it fails, so the table isn't added to the list of tables. This leads to an inconsistency in `RelationalSnapshotChangeEventSource#createSchemaChangeEventsForTables` where the lookup of the table by `TableId` returns a null `Table` entry and the call to `table.id()` leads to the NPE.

Instead I'm looking if I could refactor the DatabaseMetadata loop where for Oracle we could resolve the catalog name from either the `database.pdb.name` or `database.dbname` and use that in place of the `rs.getString(1)` call to avoid the NPE.  If that's doable then I'll update the PR with that fix but if not this may require a bit more of an in-depth change.  I'll let you know.

**Comment by Naros** [comment]
HI @gunnarmorling so this approach should be more ideal; let me know.

File: debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java [hunk 1]
```diff
@@ -2518,6 +2518,90 @@ public void shouldWarnAboutColumnNameLengthExceeded() throws Exception {
...
+
+            assertNoRecordsToConsume();
+        }
+        finally {
+            TestHelper.dropTable(connection, "dbz3611");
+        }
+    }
+
+    @Test
+    @FixFor("DBZ-3611")
```
Comment (gunnarmorling) [inline debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java:2564]:
```suggestion
    /**
     * database include/exclude list are not support (yet) for the Oracle connector; this test is just there to make 
     * sure that the presence of these (functionally ignored) properties doesn't cause any problems.
     */
    @Test
    @FixFor("DBZ-3611")
```

**Review by gunnarmorling — COMMENTED** [review]
LGTM; made one suggestion which I'll apply myself and merge. Thx, @Naros!

**Comment by Naros** [comment]
Thanks @gunnarmorling, I'm hoping this small change will also then make supporting multiple PDBs easier :).

**Comment by gunnarmorling** [comment]
Yes, that'd be cool!

>

**Pull request merged**